### PR TITLE
e2e: stabilize Ganache and failOnConsoleError

### PR DESCRIPTION
**Pull request by HowardBraham**
## **TL;DR**

- Some flaky tests were outputting invisible error messages coming from the Chrome console.  This PR makes them visible.
- If Ganache hit an error while closing, it could cause cascading errors in CI.

### Changes to package.json

- `sprintf-js` was already included by sub-packages, but now it's formally included

### Changes to retry.js

- Test error information is easier to read, with less irrelevant info

### Changes to ganache.js

- Converted to TypeScript
- Catch errors when the Ganache server closes, totally ignore the error if it's `Error: EBUSY: resource busy or locked, unlink` (seems to happen on Windows)

### Changes to run-e2e-test.js

- Log the XML result to the console for more convenient reading on local test runs

### Changes to driver.js

There was a major problem where the function was not working as intended.  If any test didn't declare `failOnConsoleError: false`, a console error would make the test fail, but the error was suppressed and was rather silent.  Instead, it would often cause these cascading errors:
```
Error: Server is already closing or closed.
Error: listen EADDRINUSE: address already in use 127.0.0.1:8545.
```
Because the `failOnConsoleError` was pretty broken, most tests declare `failOnConsoleError: false`, and basically skirt around the issue. It now helpfully logs console errors like this example:
```
----Received an error from Chrome----
TypeError: Cannot read properties of undefined (reading 'length')
  at chrome-extension://pdnpffjhgfnonmboacclhoodligjdnik/ui-9.js:10166:13
  at validate (chrome-extension://pdnpffjhgfnonmboacclhoodligjdnik/ui-9.js:10373:34)
---------End of Chrome error---------
-----failOnConsoleError is true------
```
In the long term, we probably want `failOnConsoleError: true` on most tests, to stop ignoring this.

### Changes to toggle-button.component.js

- Fixed a previous capitalization typo of `data-testId`.  This was causing one of the console errors that used to fail with the `failOnConsoleError` problem in `add-custom-network.spec.js`.

### Changes to networks-form.js

- Fixed two places where `url` could be `undefined` and thus `url.length` threw an error.  This was causing one of the console errors that used to fail with the `failOnConsoleError` problem in `add-custom-network.spec.js`.

## **Manual testing steps**

1. Look at the new CircleCI logs and test passes/failures
2. Look at the new local logs.  If you want to see the new error messages, revert `toggle-button.component.js` and `networks-form.js` one at a time, and run
`yarn test:e2e:single test/e2e/tests/add-custom-network.spec.js --browser chrome --debug`

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/ganache.js [hunk 1]
```diff
@@ -11,12 +11,32 @@ const defaultOptions = {
   quiet: true,
 };
 
+// This is a "static" variable that holds the default Ganache instance
+let ganacheInstanceOnDefaultPort = null;
```
[the remark this answers is not in the document]
Comment (HowardBraham) [reply test/e2e/ganache.js:15]: I thought about that, but it can't be a true singleton because _**sometimes**_ there are two Ganache servers.  Search for the word `concurrent` using "Match Whole Word" and you'll see it.  That's why I did that whole checking for default port stuff.

**Review by pedronfigueiredo — DISMISSED** [review]

File: test/e2e/ganache.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (HowardBraham) [reply test/e2e/ganache.js]: I made a bunch of changes based on your suggestions.  I think it's a lot better, but actually still not perfect (I still see it crash).  But let me know what you think you the changes so far.

**Comment by Menformen** [comment]
Hi baby

**Comment by Menformen** [comment]
I wan open can . Im horny nude

**Comment by Menformen** [comment]
How can I put a cam here? I don't quite understand how this works

**Review by legobeat — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]
